src/mod_zone_difficulty_scripts.cpp

```diff
@@ -229,13 +922,26 @@ class mod_zone_difficulty_unitscript : public UnitScript
                             int32 absorb = eff->GetAmount();
                             uint32 phaseMask = target->GetPhaseMask();
                             int matchingPhase = sZoneDifficulty->GetLowestMatchingPhase(mapId, phaseMask);
-                            if (sZoneDifficulty->ZoneDifficultyInfo[mapId][matchingPhase].Enabled && (matchingPhase != -1))
+                            int8 mode = sZoneDifficulty->NerfInfo[mapId][matchingPhase].Enabled;
+                            if (matchingPhase != -1)
                             {
-                                absorb = eff->GetAmount() * sZoneDifficulty->ZoneDifficultyInfo[mapId][matchingPhase].AbsorbNerfPct;
+                                Map* map = target->GetMap();
+                                if ((mode & MODE_NORMAL) == MODE_NORMAL)
+                                {
+                                    absorb = eff->GetAmount() * sZoneDifficulty->NerfInfo[mapId][matchingPhase].AbsorbNerfPct;
+                                }
+                                if ((mode & MODE_HARD) == MODE_HARD && sZoneDifficulty->HardmodeInstanceData[target->GetMap()->GetInstanceId()].HardmodeOn)
+                                {
+                                    if (map->IsRaid() ||
+                                        (map->IsHeroic() && map->IsDungeon()))
+                                    {
+                                        absorb = eff->GetAmount() * sZoneDifficulty->NerfInfo[mapId][matchingPhase].AbsorbNerfPctHard;
+                                    }
+                                }
                             }
-                            else if (nerfInDuel)
+                            else if (sZoneDifficulty->NerfInfo[DUEL_INDEX][0].Enabled > 0 && nerfInDuel)
```

Comment:
can this be negative? If not then you dont need the > 0 part

Reply:
0 is deactivated.
> 0 refers to any mode set in the `Enabled` column. `Enabled == 1` is normal mode, and `Enabled == 64` is Hardmode.

By using a check > 0, we make it future-proof in case more modes are added.

Reply:
which means you can delete > 0

if (sZoneDifficulty->NerfInfo[DUEL_INDEX][0].Enabled) does the same thing

Reply:
0 will always return false
anything with a value will always return true